server: expose prometheus and debug server on sql server

tbg opened the pull request:
- server: add prometheus endpoint to sql server
- server: expose debugServer via sql server http endpoint

tbg committed:
server: add prometheus endpoint to sql server

Release note: None

tbg committed:
server: expose debugServer via sql server http endpoint

This provides access to pprof, pprofui, logspy, and so forth, in the
same location.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53022)
<!-- Reviewable:end -->

nvb reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53022#-:-MF7I8CyAXvN9VSlV639:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @tbg)

___
*[pkg/server/testserver.go, line 561 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53022#-MF7H_sc3ruXaYxpP5wi:-MF7H_sc3ruXaYxpP5wj:bmszhho) ([raw file](https://github.com/cockroachdb/cockroach/blob/1710dd7e3c0c0a5ffd5eedb120dec019c90f04f0/pkg/server/testserver.go#L561)):*
> ```Go
> 			gossip:       gossip.MakeOptionalGossip(nil),
> 			grpcServer:   dummyRPCServer,
> 			recorder:     recorder,
> ```

Should we move this from `sqlServerOptionalKVArgs` to `sqlServerArgs` now that it's no longer optional?
___
*[pkg/server/testserver.go, line 684 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53022#-MF7GOj-0j-Jj99N2Xxi:-MF7GOj-0j-Jj99N2Xxj:b-fto8gx) ([raw file](https://github.com/cockroachdb/cockroach/blob/1710dd7e3c0c0a5ffd5eedb120dec019c90f04f0/pkg/server/testserver.go#L684)):*
> ```Go
> 	})
> 
> 	pgLAddr := pgL.Addr().String()
> ```

Should we be using `listen` like we do in `startListenRPCAndSQL` so that we properly update the `AdvertiseAddr` fields in the config and can then use them here?
___
*[pkg/server/testserver.go, line 698 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/53022#-MF7I1tG8nMJZiLBSvX5:-MF7I1tG8nMJZiLBSvX6:b-547eou) ([raw file](https://github.com/cockroachdb/cockroach/blob/4da3b1ada6569fdcbc9a9e8678645bbb1ad953d0/pkg/server/testserver.go#L698)):*
> ```Go
> 	args.stopper.RunWorker(ctx, func(ctx context.Context) {
> 		mux := http.NewServeMux()
> 		debugServer := debug.NewServer(args.Settings, s.pgServer.HBADebugFn())
> ```

nit: re-order with line below to match mux handler registration.


<!-- Sent from Reviewable.io -->

tbg committed:
server: move recorder out of sqlServerOptionalArgs

It is not optional - it powers the prometheus endpoint.

Release note: None

tbg committed:
server: use helper to open sql port in standalone server

This makes sure we update the address fields in the config (especially
important when they specified a zero port).

Release note: None

tbg committed:
server: use listen helper for sql server http port

Release note: None

tbg committed:
roachprod: fix some calls to cockroachNodeBinary

They were passing in an idx (0, 1, ...) where you needed an id (1, 2,
...). The result was that it would return "cockroach", which I noticed
since I don't have that in my path.

Release note: None

tbg committed:
serverccl: use httputil

Release note: None

tbg commented:
bors r=nvanbenschoten
